src/dryad-api.lisp

```diff
@@ -17,12 +17,15 @@
 
 ;;; messages between the dryad and its managed blossoms
 
+(deftype discover-strategy ()
+  '(member :BUST :HIT :STAY))
+
 (defstruct (message-discover (:include message))
   "Sent from a blossom process to a `DRYAD' to query for a list of other blossom processes to which it should send PINGs."
   (address         nil :type address)
   (id              nil :type t)
   (internal-weight nil :type real) ; NOTE: a little surprised that this isn't (REAL 0)
-  (repeat?         nil :type boolean)) 
+  (strategy        nil :type (or null discover-strategy)))
```

Comment:
why nullable?

Reply:
nullability in particular gives me less confidence that we've found all the places where we'd been relying on an implicit value for `repeat?`.

Reply:
(does the anatevka dryad implementation even make use of this value, or is it just part of the future-proofed API? that makes it extra suspect that we may not have properly initialized it everywhere in here...)

Reply:
I'm not sure why I made this nullable -- I think I saw the default and then assumed I had to, but that's seems incorrect in retrospect. I think dropping it should be fine. Like you say, the strategy itself is not used much in anatevka, but I have confirmed that it works elsewhere.